collector/lib/CollectorConfig.cpp

```diff
@@ -400,6 +406,48 @@ void CollectorConfig::HandleSinspEnvVars() {
   }
 }
 
+void CollectorConfig::HandleConfigMapString(const std::string& jsonString) {
+  sensor::CollectorConfig config;
+  auto status = google::protobuf::util::JsonStringToMessage(jsonString, &config);
+
+  if (!status.ok()) {
+    CLOG(ERROR) << "Failed to parse runtime config. Error: " << status.message();
+    CLOG(ERROR) << "ConfigMap file contents: " << jsonString;
```

Comment:
I have considered making this `FATAL`, but I don't think we want collector to go into a crashloop over this.

Reply:
It depends, if it is loaded at start-up I think it is okay to crash that users fix the config. 
I agree with you that the config should not crash collector at runtime.

Reply:
For now this is at startup, so I will continue to use `FATAL`.